src/renderer/routes/test_sequencer_panel/components/DesignBar.tsx

```diff
@@ -0,0 +1,266 @@
+import { useModalStore } from "@/renderer/stores/modal";
+import { useTestSequencerState } from "@/renderer/hooks/useTestSequencerState";
+import { filter } from "lodash";
+import { Button } from "@/renderer/components/ui/button";
+import { ACTIONS_HEIGHT } from "@/renderer/routes/common/Layout";
+import { FlaskConical, Import, LayoutGrid, Plus, Route } from "lucide-react";
+import {
+  StatusType,
+  TestSequenceContainer,
+  TestSequenceElement,
+} from "@/renderer/types/test-sequencer";
+import { useEffect, useState } from "react";
+import { getOnlyTests } from "./data-table/utils";
+import useWithPermission from "@/renderer/hooks/useWithPermission";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/renderer/components/ui/dropdown-menu";
+import { useImportSequences } from "@/renderer/hooks/useTestSequencerProject";
+import { CyclePanel } from "./CyclePanel";
+
+export type Summary = {
+  successRate: number;
+  numberOfTestRun: number;
+  numberOfTest: number;
+  numberOfSequenceRun: number;
+  numberOfSequence: number;
+  status: StatusType;
+};
+
+export function DesignBar() {
+  const { setIsImportTestModalOpen, setIsCreateProjectModalOpen } =
+    useModalStore();
+  const { elems, sequences, cycleRuns } = useTestSequencerState();
+  const { isAdmin } = useWithPermission();
+  const importSequences = useImportSequences();
+
+  const [summary, setSummary] = useState<Summary>({
+    numberOfTestRun: 0,
+    numberOfTest: 0,
+    numberOfSequenceRun: 0,
+    numberOfSequence: 0,
+    successRate: 0,
+    status: "pending",
+  });
+  useEffect(() => {
+    setSummary({
+      numberOfTestRun: getNumberOfTestRun(elems),
+      numberOfTest: getNumberOfTest(elems),
+      numberOfSequence: getNumberOfSequence(sequences),
+      numberOfSequenceRun: getNumberOfSequenceRun(sequences),
+      successRate: getSuccessRate(elems),
+      status: getGlobalStatus(cycleRuns, sequences, elems),
+    });
+  }, [elems, sequences, cycleRuns]);
```

Comment:
since all of this is derived from elems, sequences, and cycleRuns, there's 0 reason for this to be held as state at all
just use `useMemo` if you only want to recalculate it when the dependencies change.